OneCommunityGlobal/HighestGoodNetworkApp
# Shiwani Make permission for "total valid weekly summaries" number editable

# Description
<img width="724" alt="Screenshot 2023-10-19 at 1 37 12 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/1e85181c-c078-4b1d-9f50-73e9549d7f38">


## Related PRS (if any):
This frontend PR is related to the #[747](https://github.com/OneCommunityGlobal/HGNRest/pull/747) backend PR.


## Main changes explained:
- Added isEditCount to FormattedReports.jsx to check if 'totalValidWeeklySummaries' permission is preset in the permission list.
- Added 'totalValidWeeklySummaries' to permissionLabel in UserRoleTab.jsx
- Added 'Edit Total Valid Weekly Summaries' to modalInfo in RolePermissions.jsx

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user
5. go to dashboard→ Other Links → Permission Management → Manage User Permission → type any user -> Add 'Edit Total Valid Weekly Summaries' → Submit. If the chosen user does not have reports tab in header then add 'See Only Weekly Summary Reports tab' along with the permission set before.
6. log in as the user for whom the permissions were set.
7. go to dashboard → Reports -> Weekly Summaries Reports
8. Check if Total Valid Weekly Summaries is editable and also check if Total Valid Weekly Summaries is not editable when the permissions are removed.
9. Feel free to leave any comments or screenshots

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/7aaf7adc-ee0c-4a3a-b54d-809d0eaa90db



## Note:
Include the information the reviewers need to know.
